lib/Sema/CSSimplify.cpp

```diff
@@ -4129,6 +4129,14 @@ bool ConstraintSystem::repairFailures(
     if (repairByConstructingRawRepresentableType(lhs, rhs))
       break;
 
+    // If a closure is the argument of an apply expression, let's still record
+    // the contextual failure, but increase the score a bit in order to
+    // disambiguate from possible contextual failures located on the closure.
+    auto *AE = getAsExpr<ApplyExpr>(anchor);
+    if (AE && isExpr<ClosureExpr>(AE->getArg()->getSemanticsProvidingExpr())) {
+      increaseScore(SK_Fix);
```

Comment:
Cannot do this here because there would be no way to roll this change back later correctly, it should be done in `simplifyFixConstraint` while handling this fix kind.

Reply:
Ah makes sense, thank you!
I'll check this out

Reply:
Sounds good!